stable/app/templates/poddisruptionbudget.yaml

```diff
@@ -0,0 +1,12 @@
+{{- if .Values.maxUnavailable }}
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: "{{ template "app.fullname" . }}-pdb"
+spec:
+  maxUnavailable: {{ .Values.maxUnavailable }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: "{{ template "app.fullname" . }}"
```

Comment:
here we need to select pod labels, and. NOT the deployment

Reply:
The same pod labels are applied to db migration jobs as well which caused the issue of incompatibility. And required additional/differentiating labels.

Reply:
should we add new labels to selector like:

```
app: "{{ template "app.fullname" . }}"
```
Incase there is a overlap ?

Reply:
Yes. However adding labels to the pods result in incompatibility since labels are immutable and cannot be changed for existing deployment

Reply:
I think we have the issue of overlapping labels between the:
- Pods owned by `Deployment`
- Pods owned by migration-job `Job`

Would it make sense to add a static label to the `Deployment`'s and `PDB`'s matchSelector with something like `app.kubernetes.io/governed-by: deployment`

This would make sure that `PDB` won't be over-sighting any pods owned by controllers other than deployment.

Reply:
For reference some commonly used labels -
https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/
https://kubernetes.io/docs/reference/labels-annotations-taints/

Reply:
> Would it make sense to add a static label to the Deployment's and PDB's matchSelector  

Correction here, since `deploy`'s `spec.selector.matchLabels` is an immutable field, this should ideally go as an additional field to the `spec.template.metadata.labels` field (pod's labels) – which can then be matched by the `pdb`'s `selector.matchLabels`

Reply:
also I would highly suggest using custom labels like `app.pixxel.co.in/governed-by` instead of `app.kubernetes.io/governed-by` to easily distinguish between our organisation specific labels vs those applied by k8s tooling internally or other cloud native tools

Reply:
Additional labels would be going through the chart's values.yaml and not in the chart spec itself.
Here, we require a unique label which distinguishes chart's deployment pods from deployment's migration-job and other pods deployed by cron for the same Helm release. Thus, a k8s standard label would make more sense, since it's not intrinsic to the app chart and it's PDB. Rather than any external metadata which the user want to add.

Reply:
Yes if possible can we use one of the common labels here - 
```
app.kubernetes.io/name
app.kubernetes.io/instance
app.kubernetes.io/version
app.kubernetes.io/component
app.kubernetes.io/part-of
app.kubernetes.io/managed-by
```
For example `app.kubernetes.io/component=server` ?

Reply:
Let's avoid `app.pixxel.co.in` because this is not a pixxel specific use case that we are solving with this label.